### PR TITLE
Deprecated Cache lib

### PR DESCRIPTION
**Pull request by Toflar**
Marking `Contao/Cache` as deprecated. It's a very simple in-memory cache which is better implemented in your own classes. Persistence was never supported so there's no real point in maintaining this class.
If you need persistence, use `symfony/cache` instead.

### CODE REVIEW

**Comment by aschempp** [comment]
I think we should rewrite the usage of Cache (from https://github.com/contao/contao/pull/4330) in 4.13 to already deprecated it.

**Comment by Toflar** [comment]
Yeah, I didn't bother because 🤷‍♂️

**Review by ausi — DISMISSED** [review]

File: comments-bundle/src/Resources/contao/dca/tl_comments.php [hunk 1]
```diff
@@ -387,16 +386,18 @@ protected function isAllowedToEditComment($intParent, $strSource)
 			return true;
 		}
 
+		static $cache = array();
```
Comment (leofeyer) [inline comments-bundle/src/Resources/contao/dca/tl_comments.php:389]: This is considered bad practice (see https://twitter.com/s_bergmann/status/1182282457011363842). If we go this way, it should be a static class property. However, I am not convinced about the change.

Comment (leofeyer) [inline comments-bundle/src/Resources/contao/dca/tl_comments.php:389]: This is considered bad practice (see https://twitter.com/s_bergmann/status/1182282457011363842). If we go this way, it should be a static class property. However, I am not convinced about the change.

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Review by leofeyer — CHANGES_REQUESTED** [review]
I don‘t see any advantage in removing the class.

**Review by aschempp — DISMISSED** [review]

**Review by fritzmg — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**